### PR TITLE
print device when it's not on default device

### PR DESCRIPTION
**Pull request by ailzhang**
we used to not print device when it's on xla. It's sometimes confusing as it looks the same as cpu tensor...

### CODE REVIEW

File: torch/_tensor_str.py [hunk 1]
```diff
@@ -239,9 +239,17 @@ def get_summarized_data(self):
...
-    if not torch._C._is_default_type_cuda():
+
+    # Note [Print tensor device]:
+    # A general logic here is we only print device when it doesn't match
+    # the device specified in default tensor type.
+    # Currently torch.set_default_tensor_type() only supports CPU/CUDA.
+    # In other cases, we don't have a way to set them as default yet,
+    # and we should always print out device for them.
+    if self.device.type == 'xla':
+        suffixes.append('device=\'' + str(self.device) + '\'')
```
Comment (SsnL) [inline torch/_tensor_str.py]: test? :)

Comment (ailzhang) [reply torch/_tensor_str.py]: there isn't a way to trigger it in pytorch repo. I will add one in pytorch/xla once this is in. :D

Comment (SsnL) [reply torch/_tensor_str.py]: awesome!

File: torch/_tensor_str.py [hunk 2]
```diff
@@ -239,9 +239,17 @@ def get_summarized_data(self):
...
+    elif not torch._C._is_default_type_cuda():
```
Comment (apaszke) [inline torch/_tensor_str.py]: Perhaps we could have a method that queries the default device instead, and prints this block if the tensor device is not that one? That would make it robust to all kinds of choices, and we wouldn't have to revisit this code in the future.

**Comment by ailzhang** [comment]
@apaszke I changed the api to return default device string instead of a bool to tell whether it's cuda. Since this is the only usage of that (internal) api I think it's okay to just change it.  Please let me know if this looks good. Thanks!

**Review by apaszke — APPROVED** [review]

**Pull request closed without merging**